r4/resource-apis/uscore501-patient/uscore501_patient_connect.bal

```diff
@@ -21,73 +21,111 @@
 
 import ballerinax/health.fhir.r4;
 import ballerinax/health.fhir.r4.uscore501;
+import ballerina/http;
+import ballerinax/health.fhir.r4.parser;
 
-public isolated class Uscore501PatientSourceConnect {
+configurable string sourceSystem = "http://localhost:9595/uscore";
+
+final string READ = sourceSystem.endsWith("/") ? "read/" : "/read/";
+final string SEARCH = sourceSystem.endsWith("/") ? "search" : "/search";
+final string CREATE = sourceSystem.endsWith("/") ? "create" : "/create";
+
+final http:Client sourceEp = check new (sourceSystem);
+
+public isolated class UscorePatientSourceConnect {
 
     *PatientSourceConnect;
     isolated function profile() returns r4:uri {
         return "http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient";
     }
 
-    isolated function read(string id, r4:FHIRContext ctx) returns Patient|r4:FHIRError {
-
-        //Implement source system connection here and retreive data.
-        //Create FHIR resource from retreived data.
-        uscore501:USCorePatientProfile example = {
-            id: "12d39",
-            meta: {
-                versionId: "abc12s3",
-                profile: ["http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"]
-            },
-            identifier: [
-            ],
-            implicitRules: "https://www.hl7.org/fhir",
-            language: "en-US",
-            gender: "unknown", 
-            name: []};
-        return example;
+    isolated function read(string id, r4:FHIRContext fhirContext) returns Patient|r4:FHIRError {
+
+        http:Response|http:ClientError res =  sourceEp->get(READ + id);
+        if (res is http:ClientError) {
+            r4:FHIRError fhirError = r4:createFHIRError("Error occured when calling the source system.", r4:CODE_SEVERITY_ERROR,r4:TRANSIENT_EXCEPTION);
+            return fhirError;
+        } else {
+            json|error payload = res.getJsonPayload();
+            if (payload is error) {
+                r4:FHIRError fhirError = r4:createFHIRError("Unable to extract JSON payload from the source response.", r4:CODE_SEVERITY_ERROR,r4:TRANSIENT_EXCEPTION);
+                return fhirError;
+            } else {
+                uscore501:USCorePatientProfile|error fhirResource = <uscore501:USCorePatientProfile>check parser:parse(payload, uscore501:USCorePatientProfile);
```

Comment:
Do we need this typecast? We didn't need one when the cloneWithType was used.

Reply:
Yes. unless there will be an error saying `incompatible types`.

Reply:
Then IMO we should handle that instead of the typecast. Ideally, it should be like `uscore501:USCorePatientProfile|error fhirResource = parser:parse(payload, uscore501:USCorePatientProfile);`. No check and no typecast. But handle the error.

Reply:
Since the function definition for `parse` looks as follows, can we get the expected record without typecast? (returns `anydata|r4:FHIRParseError`)
``` 
public isolated function parse(json|xml payload, typedesc<anydata>? targetFHIRModelType = ()) returns anydata|r4:FHIRParseError 
```